OvmfPkg/PlatformPei/PlatformId.c

```diff
@@ -0,0 +1,660 @@
+    DEBUG ((DEBUG_ERROR, DPREFIX "RimLocatorType %d\n", Status));
+    return Status;
+  }
+
+  // UINT32 RimLocatorLength;
+  // UINT8  RimLocator[RimLocatorLength];
+  Status = ConsumeUint32Str (Cursor, BufferEnd, &Cursors->RimLocator, &Cursors->RimLocatorEnd);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "RimLocator %d\n", Status));
+    return Status;
+  }
+
+  Status = ValidateLocator (
+             Cursors->RimLocator,
+             Cursors->RimLocatorEnd,
+             Cursors->RimLocatorType
+             );
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  // UINT32 PlatformCertLocatorType;
+  Status = ParseUint32 (Cursors->RimLocatorEnd, BufferEnd, &Cursors->PlatformCertLocatorType, &Cursor);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "PlatformCertLocatorType %d\n", Status));
+    return Status;
+  }
+
+  // UINT32 PlatformCertLocatorLength;
+  // UINT8  PlatformCertLocator[PlatformCertLocatorLength];
+  Status = ConsumeUint32Str (Cursor, BufferEnd, &Cursors->PlatformCertLocator, &Cursors->PlatformCertLocatorEnd);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, DPREFIX "PlatformCertLocator %d\n", Status));
+    return Status;
+  }
+
+  return ValidateLocator (
+           Cursors->PlatformCertLocator,
+           Cursors->PlatformCertLocatorEnd,
+           Cursors->PlatformCertLocatorType
+           );
+}
+
+/**
+ * Converts an octet string between Start and End to its NUL-terminated
+ * hexadecimal string representation.
+ *
+ * @param[out] Buffer  The buffer to write the hexadecimal string to.
+ * @param[in]  Size  The size of the buffer.
+ * @param[in]  Start  The start of the octet string.
+ * @param[in]  End  The end of the octet string.
+**/
+STATIC
+VOID
+BytesToHex (
+  OUT CHAR8  *Buffer,
+  IN UINTN   Size,
+  IN UINT8   *Start,
+  IN UINT8   *End
+  )
+{
+  UINT8  Byte;
+  UINTN  Index   = 0;
+  UINTN  Nibbles = 2 * (UINTN)(End - Start);
+
+  for (Index = 0; Index < Size-2 && Index < Nibbles; Index += 2) {
+    Byte            = Start[Index/2];
+    Buffer[Index]   = HEX_DIGITS[Byte >> 4];
+    Buffer[Index+1] = HEX_DIGITS[Byte & 0x0F];
+  }
+
+  Buffer[Index] = 0;
+}
+
+/**
+ * Converts an octet string between Start and End to a NUL-terminated C string.
+ *
+ * PREREQUISITE: Start <= End
+ *
+ * @param[out] Dest  The buffer to write the C string to.
+ * @param[in]  DestSize  The size of the buffer.
+ * @param[in]  Start  The start of the octet string.
+ * @param[in]  End  The end of the octet string.
+**/
+STATIC
+VOID
+ByteStrToCstr (
+  OUT CHAR8  *Dest,
+  IN  UINTN  DestSize,
+  IN  UINT8  *Start,
+  IN  UINT8  *End
+  )
+{
+  UINTN  Size = (UINTN)(End - Start);
+
+  if (Size > DestSize - 1) {
+    Size = DestSize - 1;
+  }
+
+  CopyMem (Dest, Start, Size);
+  Dest[Size] = 0;
+}
+
+STATIC VOID
+DebugLocator (
+  IN UINTN   DebugLevel,
+  IN UINT8   *Locator,
+  IN UINT8   *LocatorEnd,
+  IN UINT32  LocatorType
+  )
+{
+  CHAR8  Buffer[256];

```

Comment:
why hardcode 256?

Reply:
Removed for uninterpreted single file solution.